crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -437,7 +458,9 @@ pub enum VerificationError {
     #[error(
         "invalid internalization: solution attempts to internalize tokens which are not trusted"
     )]
-    Internalization,
+    UntrustedInternalization,
+    #[error("invalid internalization: solution attempts to internalize failing interactions")]
```

Comment:
Not sure if nit picking is worth it here but technically the simulation failure could originate from an interaction that is not supposed to be internalized. I think saying that the uninternalized version of the settlement failed to simulate is specific enough.

Reply:
An example of this would be:
- Interaction 1: Swap 50 tokens over pool A _internalized_
- Interaction 2: Swap 50 tokens over pool A _NON-internalized_

In this example, a solver would split the swaps over the same pool in order to get better prices (since there would be less price impact). Each on their own simulates successfully, and here, the second interaction is the one to revert.